packages/circuits/tests/unpack-element.test.ts

```diff
@@ -0,0 +1,47 @@
+import { WitnessTester } from "circomkit"
+import { circomkit, genRandomBabyJubValue, getSignal } from "./common"
+
+describe("unpack-element", () => {
+    let circuit: WitnessTester<["in"], ["out"]>
+
+    it("Should unpack a field element with 3 / 4 / 5 packed values correctly", async () => {
+        for (let n = 3; n <= 5; n += 1) {
+            const elements: string[] = []
+
+            circuit = await circomkit.WitnessTester("unpack-element", {
+                file: "unpack-element",
+                template: "UnpackElement",
+                params: [n]
+            })
+
+            for (let i = 0; i < n; i += 1) {
+                let e = (BigInt(genRandomBabyJubValue().toString()) % BigInt(2 ** 50)).toString(2)
+
+                while (e.length < 50) {
+                    e = `0${e}`
+                }
+
+                elements.push(e)
+            }
+
+            const INPUT = {
+                in: BigInt(`0b${elements.join("")}`)
+            }
+
+            const witness = await circuit.calculateWitness(INPUT)
+            await circuit.expectConstraintPass(witness)
+
+            const output = []
+            for (let i = 0; i < n; i += 1) {
+                const out = await getSignal(circuit, witness, `out[${i}]`)
```

Comment:
I'm not super sure about how this works. Just wanted to double-check. We are getting the output generated with `calculateWitness` right?

If so, doesn't the `expectPass` test always pass? Shouldn't it use an explicit value as the expected output rather than calculating it?

Reply:
See [8027bf7](https://github.com/privacy-scaling-explorations/zk-kit/pull/163/commits/8027bf7d81f4b5c207ef025d83e72195776f4cc9)